bn-apps/memberships-management/membership-service/src/main/kotlin/com/r3/businessnetworks/membership/flows/bno/SelfIssueMembershipFlow.kt

```diff
@@ -2,60 +2,109 @@ package com.r3.businessnetworks.membership.flows.bno
+        progressTracker.currentStep = ACTIVATING_MEMBERSHIP
+            val txBuilder = TransactionBuilder(notary)
+                .addInputState(bnoMembership)
+                //Second transaction will only modify the status of the BNO node and set it to ACTIVE
+                .addOutputState(bnoMembership.state.data.copy(
+                                status = MembershipStatus.ACTIVE,
+                                modified = serviceHub.clock.instant()),
+                                MembershipContract.CONTRACT_NAME)
+                .addCommand(MembershipContract.Commands.Activate(), ourIdentity.owningKey)
+            txBuilder.verify(serviceHub)
+            val stx = serviceHub.signInitialTransaction(txBuilder)
+            // sign the transaction so it can be written to the ledger
+            subFlow(FinalityFlow(stx, listOf())) //listOf remains empty since only BNO needed to sign the transaction
+            logger.info("Membership has been activated")
+            progressTracker.currentStep = ACTIVATED_MEMBERSHIP
+            return subFlow(FinalityFlow(stx, listOf()))
+            //end of second transaction to set membership status to ACTIVE
+    }
+}
+
+/**
+ * This is a convenience flow that can be easily used from a command line
+ *
+ * @param party whose membership state to be activated
+ */
+@InitiatingFlow
+@StartableByRPC
+open class ActivateBnoMembershipFlow(val party: Party) : BusinessNetworkOperatorFlowLogic<SignedTransaction>() {
 
-        progressTracker.currentStep = RequestingMembership
-        val stx = subFlow(RequestMembershipFlow(ourIdentity, metaData))
+    companion object {
+        object LOOKING_FOR_MEMBERSHIP_STATE : ProgressTracker.Step("Looking for party's membership state")
+        object ACTIVATING_THE_MEMBERSHIP_STATE : ProgressTracker.Step("Activating the membership state")
 
-        val outState = stx.tx.outputs.single()
-        val membershipState = outState.data as MembershipState<*>
+        fun tracker() = ProgressTracker(
+                LOOKING_FOR_MEMBERSHIP_STATE,
+                ACTIVATING_THE_MEMBERSHIP_STATE
+        )
+    }
 
-        // If auto activate is set then don't need to call ActivateMembership
-        return if (!membershipState.isActive()) {
-            logger.info("Membership is not yet active")
-            progressTracker.currentStep = ActivatingMembership
-            val output = stx.tx.outRefsOfType(MembershipState::class.java).single()
+    override val progressTracker = tracker()
 
-            subFlow(ActivateMembershipFlow(output))
-        } else {
-            logger.info("Membership was automatically activated")
-            stx
-        }
+    @Suspendable
+    override fun call(): SignedTransaction {
+        progressTracker.currentStep = ACTIVATING_THE_MEMBERSHIP_STATE
+        val stateToActivate = findMembershipStateForParty(party)
+        return subFlow(SelfIssueMembershipFlow(stateToActivate))
```

Comment:
Why are you passing an existing membership states as a metadata to issue a *new* membership?

Reply:
sorry that should have been removed from the newer version, that was from taken from the ActivateMebershipFlow and I was going to change it to fit the older version of SelfIssueMembershipFlow, will remove this now.